diff --git a/pyproject.toml b/pyproject.toml
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -47,7 +47,8 @@ dependencies = [
   "torch>=1.13.1",
   "torch>=2.6.0; python_version >= '3.12'",
   "torchvision>=0.14.1,<0.20; python_version < '3.9'",
-  "torchvision>=0.17.0; python_version >= '3.9'",
+  "torchvision>=0.14.1; python_version >= '3.9' and python_version < '3.12'",
+  "torchvision>=0.17.0; python_version >= '3.12'",
   "tqdm",
   "typing-extensions>=4.0.0",
   "wheel>=0.32.0", # For --no-build-isolation.
